### ISSUE TITLE
"checksig" test always pass even when it should not

### ISSUE THREAD

**Issue by binaryFate** [monero-project/mininero#1]
The "checksig" test in mininero.py always pass, even if the private key is arbitrarily changed. 

For instance, when changing the private key before signing, like this on line 700:
sir, sic = generate_signature(prefix, pub,
 99999999999999999999999999999999999999999999999999999999999999999999999999999)

check_signature() still returns True.

**Comment by ghost** [comment]
Dear binaryfate, I haven't updated this code in a while - recent version of my code (which should be more compatible with actual Monero) is at https://github.com/ShenNoether/MiniNero - sometime in the near future, I will most likely replace the code here with that, however I'm quite busy with the Ring CT stuff, as well as some unrelated projects, so it might be a month or two (iirc the original MiniNero code was just for the purpose of seeing how the ring sigs worked, and I didn't bother with the basic stuff)

**Comment by binaryFate** [comment]
Yes, I had a look at your more recent version and it's indeed much cleaner. Very appreciated!
I must say I miss the tests there though, many tests are in old version and not in new one. They're very useful in learning how to do stuff.

**Comment by ghost** [comment]
Sure, once I get the ringCT stuff finished, I can add those tests into 
the newer version.
